cdap-watchdog/src/main/java/co/cask/cdap/metrics/collect/MessagingMetricsCollectionService.java

```diff
@@ -161,6 +183,11 @@ void publish(MessagingService messagingService) throws IOException {
               done = true;
             }
           }
+        } catch (IOException ioe) {
+          String exceptionMessage =
+            String.format("Exception while publishing metrics for tags: %s to topic: %s with payload size: %s bytes",
+                          MAP_JOINER.join(logContexts.asMap()), topicId.getTopic(), payloadSize);
```

Comment:
what will this look like in the actual message? Do you have an example?

Reply:
can you also include the total number of metrics it tried to emit?
That would tell the user: hey your program xyz attempted to emit 1500 metrics with a total size of 1456732645 bytes.